### DISCUSSION TITLE
Language Servers

### DISCUSSION THREAD

**Discussion by gr9** [phcode-dev/phoenix#587] [General]
Howdy!

First of all - was pleasantly surprised to see brackets being picked up by phoenix; so thanks to all involved.  Just wondering if phoenix brings any more language servers to the table? E.g. compiler support, hints, autocomplete etc for other programming languages.  Compared to vs code etc, this is one area I've found brackets lacking in the past.  Just did a search for language servers re: phoenix & didn't find anything as of yet.

**Comment by abose** [comment]
Currently, most of our development is for attaining feature parity with brackets. Over the next month we will be concentrating on enabling code intelligence for Javascript in Phoenix which will eventually be available in Brackets as well.

After that language server support is also in the pipeline after enabling extension support.

If you have any suggestions or use cases for js vode hints, just drop here, will check out while we work on thr new code intelligence framework.

**Reply by gr9** [reply]
Oh ok.. makes sense.. was just wondering if there was any new phoenix code that brackets might benefit from in the form of language server implementations.  Feature parity seems like a sensible priority though.

I don't have any specific requests, just curious.  In my personal case I've been interested in getting my hands on a Java language server implementation for brackets.  Given the other very java specific IDE's out there though I can see why this is probably a bit of a niche request.  In general though, more language server implementations obviously makes an IDE more versatile / appealing to a wider audience..  Was pumped when I saw brackets announce that it now supports language features (a few years ago), only to see very few implementations become available.

Might have a crack at my own implementation time permitting ;)

Thanks for the quick reply.

**Reply by abose** [reply]
Would be cool to have more language support in core. We will be working on improving language server support in core. Once the support is there, i guess it would be much easier to add language support like java and others. 

For any help, just ping in discord or GitHub issues.
